ui/base-tool/src/main/java/org/cru/godtools/base/tool/databinding/TextViewAdapters.kt

```diff
@@ -3,9 +3,44 @@ package org.cru.godtools.base.tool.databinding
 import android.widget.TextView
 import androidx.annotation.ColorInt
 import androidx.databinding.BindingAdapter
+import com.squareup.picasso.Picasso
+import org.ccci.gto.android.common.picasso.widget.TextViewDrawableEndTarget
+import org.ccci.gto.android.common.picasso.widget.TextViewDrawableStartTarget
+import org.cru.godtools.base.fileManager
 import org.cru.godtools.base.tool.model.view.bindTo
+import org.cru.godtools.xml.model.Resource
 import org.cru.godtools.xml.model.Text
 
 @BindingAdapter("android:text", "android:textSize", "defaultTextColor", requireAll = false)
 fun TextView.bindTextNode(text: Text?, textSize: Float?, @ColorInt defaultTextColor: Int?) =
     text.bindTo(this, textSize, defaultTextColor)
+
+@BindingAdapter("android:drawableStart", "startImageSize")
+fun TextView.bindDrawableStartResource(resource: Resource?, startImageSize: Int) {
+    val target = TextViewDrawableStartTarget.of(this)
+    val file = resource?.localName?.let { context.fileManager.getFileBlocking(it) }
+    val imageSize = (startImageSize * resources.displayMetrics.density).toInt()
```

Comment:
I like using `TypedValue` to do the conversion for us instead of hard-coding the formula.

```
TypedValue.applyDimension(COMPLEX_UNIT_DIP, startImageSize, resources.displayMetrics)
```

Reply:
At some point I'm going to play around with the layout a little bit to see if data binding does any automatic conversions for us in the case of using a `dimen` value directly for the size attribute.

If so I might adjust where we do the dp -> px conversion to better mesh with data binding

Reply:
@frett So to fix the typing issues this is what I have
```
val imageSize = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, startImageSize.toFloat(), resources.displayMetrics).toInt()
```

Does that look alright to you?  ```applyMethod``` expects a ```Float``` for the second argument so it's kind of weird to have to turn the Int -> Float and then convert the whole thing to an Int.

Reply:
hmm, let me do a bit of research

Reply:
stick with what you have for the time being, I might revisit this later and implement some Kotlin utils to make something like this possible:

```kotlin
Dp(startImageSize).toPx(resources.displayMetrics).toInt()
```